tests/e2e-tests/utils/components.js

```diff
@@ -0,0 +1,130 @@
+		await expect( page ).toFill( `input[name="attribute_names[${ i }]"]`, 'attr #' + ( i + 1 ) );
+		await expect( page ).toFill( `textarea[name="attribute_values[${ i }]"]`, 'val1 | val2' );
+		await expect( page ).toClick( `input[name="attribute_variation[${ i }]"]` );
+	}
+
+	await expect( page ).toClick( 'button', { text: 'Save attributes' } );
+
+	// Wait for attribute form to save (triggers 2 UI blocks)
+	await uiUnblocked();
+	await uiUnblocked();
+
+	// Create variations from attributes
+	await clickTab( 'Variations' );
+	await page.waitForSelector( 'select.variation_actions:not([disabled])' );
+	await page.focus( 'select.variation_actions' );
+	await expect( page ).toSelect( 'select.variation_actions', 'Create variations from all attributes' );
+
+	const firstDialog = await expect( page ).toDisplayDialog( async () => {
+		// Using this technique since toClick() isn't working.
+		// See: https://github.com/GoogleChrome/puppeteer/issues/1805#issuecomment-464802876
+		page.$eval( 'a.do_variation_action', elem => elem.click() );
+
+	} );
+
+	expect( firstDialog.message() ).toMatch( 'Are you sure you want to link all variations?' );
+
+	const secondDialog = await expect( page ).toDisplayDialog( async () => {
+		await firstDialog.accept();
+	} );
+
+	expect( secondDialog.message() ).toMatch( '8 variations added' );
+	await secondDialog.dismiss();
+
+	// Set some variation data
+	await uiUnblocked();
+	await uiUnblocked();
+
+	await page.waitForSelector( '.woocommerce_variation .handlediv' );
+
+	await expect( page ).toClick( '.woocommerce_variation:nth-of-type(1) .handlediv' );
+	await page.focus( 'input[name="variable_is_virtual[0]"]' );
+	await expect( page ).toClick( 'input[name="variable_is_virtual[0]"]' );
+	await expect( page ).toFill( 'input[name="variable_regular_price[0]"]', '9.99' );
+
+	await expect( page ).toClick( '.woocommerce_variation:nth-of-type(2) .handlediv' );
+	await page.focus( 'input[name="variable_is_virtual[1]"]' );
+	await expect( page ).toClick( 'input[name="variable_is_virtual[1]"]' );
+	await expect( page ).toFill( 'input[name="variable_regular_price[1]"]', '11.99' );
+
+	await expect( page ).toClick( '.woocommerce_variation:nth-of-type(3) .handlediv' );
+	await page.focus( 'input[name="variable_manage_stock[2]"]' );
+	await expect( page ).toClick( 'input[name="variable_manage_stock[2]"]' );
+	await expect( page ).toFill( 'input[name="variable_regular_price[2]"]', '20' );
+	await expect( page ).toFill( 'input[name="variable_weight[2]"]', '200' );
+	await expect( page ).toFill( 'input[name="variable_length[2]"]', '10' );
+	await expect( page ).toFill( 'input[name="variable_width[2]"]', '20' );
+	await expect( page ).toFill( 'input[name="variable_height[2]"]', '15' );
+
+	await page.focus( 'button.save-variation-changes' );
+	await expect( page ).toClick( 'button.save-variation-changes', { text: 'Save changes' } );
+
+	await verifyAndPublish();
+};
+
+export { createSimpleProduct };
```

Comment:
We're going to get conflicts when we merge things into the main branch, since this file in particular is duplicated e.g. https://github.com/woocommerce/woocommerce/pull/24670/files#diff-a39ea7ab3ac6262c137fbcb467aed169

Reply:
Resolved conflicts for now.